charts/dns/templates/pdns-deployment.yaml

```
@@ -6,65 +6,125 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
   name: {{ include "pdns.fullname" . }}
+  namespace: {{ .Release.Namespace }}
   labels:
-    {{- include "pdns.labels" . | nindent 4 }}
+    {{- include "stateless-dns.labels" . | nindent 4 }}
+    {{- include "pdns.selectorLabels" . | nindent 4 }}
 spec:
+  replicas: {{ .Values.replicaCount }}
   selector:
     matchLabels:
       {{- include "pdns.selectorLabels" . | nindent 6 }}
+  {{- with .Values.deploymentStrategy }}
+  strategy:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
   template:
     metadata:
-      {{- with .Values.podAnnotations }}
+      
       annotations:
+        checksum/apikey-secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
```

I still get nervous when I see secrets hashed with bruteforceable algorithms. So I'm going to get wild here:

```suggestion
        checksum/apikey-secret: {{ include (print $.Template.BasePath "/secret.yaml") . | bcrypt }}
```

---

Fair. Done.